src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -104,6 +104,13 @@ protected TestClass createTestClass(Class<?> testClass) {
     protected abstract Description describeChild(T child);
 
     /**
+     * Fires test ignore through {@code notifier} for child corresponding to
+     * {@code child}, which can be assumed to be an element of the list returned
+     * by {@link ParentRunner#getChildren()}.
+     */
+    protected abstract void fireIgnoreChild(T child, RunNotifier notifier);
```

Comment:
It is possible to be compatible with a default implementation of fireIgnoreChild(), which calls runChild():
* The class BlockJUnit4ClassRunner is able to override both methods as intended. This means, new implementations have not to check, if the given child is ignored or not.
* Old implementations will have the new improved behavior, because they test already itself, if the given child is ignored or not.